### ISSUE TITLE
IBus Error

### ISSUE THREAD

**Issue by moore-bryan** [pop-os/cosmic-epoch#1243]
Running cosmic-epoch from ryanabx's copr repo on Fedora 41 and I keep getting this error upon login, but none of the solutions work; that is, explicitly setting "IBUs Wayland" in KDE does nothing and org.freedesktop.IBus.Panel doesn't exist, let alone that which comes after "Panel."

Any ideas?
![screenshot-2024-11-12-18-51-42](https://github.com/user-attachments/assets/91af14cc-16af-438f-a7cc-90d9c29c4664)

**Comment by git-f0x** [comment]
COSMIC doesn't use IBus, so maybe that's why the notification comes up (on Pop, IBus is disabled).

**Comment by moore-bryan** [comment]
Hmm... What does COSMIC use "instead?"

**Comment by gssilva57** [comment]
Same error here on Fedora 42 COSMIC

**Comment by badlydrawnface** [comment]
Ditto, Fedora 42.

**Comment by daholland** [comment]
same issue  on a freshly updated fedora 41:

I saw this issue too which seems related https://github.com/pop-os/cosmic-session/issues/110

**Comment by shohan11d** [comment]
same on fedora 42. not fixed yet

**Comment by zarugo** [comment]
You see the notification because the package `imsettings` installed by default creates a dbus service (com.redhat.imsettings) that I guess gets called by someone, and this triggers the execution of `/usr/libexec/imsettings-daemon` that, in its turn, launches the ibus daemon.

TL;DR: if you uninstall imsettings, I don't think that there are drawbacks in Cosmic, and the notification will be gone.

If you are curious and want to investigate:

The definition of the dbus service:

`cat /usr/share/dbus-1/services/com.redhat.imsettings.service`

To see the transient unit that is launched:

 `systemctl --user list-units --type service | grep imsettings`

then, once you have the name (it can vary since it's created at runtime), cat it, for example:

`systemctl --user cat dbus-:1.2-com.redhat.imsettings@0.service`

I think that the notification is complaining that iBus is being launched in the wrong way. Not an expert, but, at the very least, the [documentation](https://github.com/ibus/ibus/wiki/WaylandDesktop#cosmic-dekstop-enviroment-1) says that, in Cosmic, it should be launched with `ibus start --type wayland` and  (even if I don't think that that's the specific reason for the notification complaints) if you `ps aux | grep ibus` you see that imsettings launches it somewhat differently, `ibus-daemon -r --xim`, that is not the same you get if you do `ibus start --type wayland`

**Comment by chdalski** [comment]
More informations about IMSettings can be found here: [IMSettings - Fedora Wiki](https://fedoraproject.org/wiki/User:Tagoh/IMSettings)

It would also be possible to update the settings `/etc/X11/xinit/xinput.d/ibus.conf`.
I didn't try and uninstalled `imsettings`, but `NOT_RUN` or `IMSETTINGS_IGNORE_SESSION` looks promising.

If someone wants to experiment with it I would suggesting going with `IMSETTINGS_IGNORE_SESSION=COSMIC-wayland`.

**Comment by icecreammatt** [comment]
> More informations about IMSettings can be found here: [IMSettings - Fedora Wiki](https://fedoraproject.org/wiki/User:Tagoh/IMSettings)
> 
> It would also be possible to update the settings `/etc/X11/xinit/xinput.d/ibus.conf`. I didn't try and uninstalled `imsettings`, but `NOT_RUN` or `IMSETTINGS_IGNORE_SESSION` looks promising.
> 
> If someone wants to experiment with it I would suggesting going with `IMSETTINGS_IGNORE_SESSION=COSMIC-wayland`.

Thank you for this info! This fixed that message from popping up for me when I switched it. I've been dealing with that popup for months ever since switching to Cosmic desktop on Asahi Linux.

`IMSETTINGS_IGNORE_SESSION=KDE-wayland` to `IMSETTINGS_IGNORE_SESSION=COSMIC-wayland`

---

Edit: 9/15 The error is back again after updating the other day. I'm no longer using Cosmic for about a month now but I suspect some components of it are still installed which are triggering this alert.

Updating this line seems to have fixed it.
```
# IMSETTINGS_IGNORE_SESSION=COSMIC-wayland
IMSETTINGS_IGNORE_SESSION=niri-wayland
```

**Comment by chawyehsu** [comment]
Thank you all for the investigation! I'm on Fedora 43 and don't use IBus. I managed to get rid of the annoying notification popup by updating the `/etc/X11/xinit/xinput.d/ibus.conf` file. What changes I made:

```ini
IMSETTINGS_IGNORE_SESSION=*-wayland ; was KDE-wayland
```

You can actually use a wildcard to match all types of wayland session in any case you might switch to. I found this by inspecting `/etc/X11/xinit/xinput.d/none.conf`. And detailed parameters documentation of `imsettings` can be found on https://gitlab.com/tagoh/imsettings.

If you don't use the `imsettings` feature - change IMF on demand, and yet want to keep the package installed, then it is possible to disable it by setting `DISABLE_IMSETTINGS` according to the doc mentioned above. I haven't tried it though.
